meta-ros-common/recipes-extended/geographiclib/geographiclib_1.48.bb

```diff
@@ -11,4 +11,24 @@ SRC_URI[sha256sum] = "7203d56123b6f6fb31842295d57b2418f79fb0db9a06f2f65ee9e415c6
 
 S = "${WORKDIR}/GeographicLib-${PV}"
 
+PROVIDES += "${PN}-tools"
```

Comment:
This doesn't seem to be needed anywhere.

Reply:
Wouldn't this allow the generator to detect that `geographiclib-tools` is available instead of adding as `ROS_UNRESOLVED_PLATFORM_PKG_geographiclib-tools = "geographiclib-tools"`?

Reply:
If something needs only the tools in runtime, then it should set just
RDEPENDS_${PN} = "geographiclib-tools"
and bitbake will find geographiclib recipe which provides this package correctly.

The issue is this line you referenced in package.xml
<build_depend>geographiclib-tools</build_depend>

which adds buildtime dependedncy on geographiclib-tools _package_ to DEPENDS. There is already build time dependency on geographiclib in package.xml:
<build_depend>geographiclib</build_depend>

geographiclib-tools should just be removed from DEPENDS in .bbappend, adding PROVIDES just to work around an issue in package.xml doesn't really improve that.